contrib/pkg/provision/command.go

```diff
@@ -0,0 +1,206 @@
+	RPMRepositoryURL    string
+	RPMRepositorySource string
+	SSHSecretName       string
+	AccountSecretName   string
+	KeyPairName         string
+	AnsibleImage        string
+	Region              string
+	KubeconfigFile      string
+	Version             string
+	VPCName             string
+	LogLevel            string
+
+	ImageID string
+	Logger  log.FieldLogger
+}
+
+func NewProvisionClusterCommand() *cobra.Command {
+	userName := "user"
+	u, err := user.Current()
+	if err == nil {
+		userName = u.Username
+	}
+
+	opt := &ProvisionClusterOptions{}
+	cmd := &cobra.Command{
+		Use:   "provision NAME",
+		Short: "Provision a new cluster with the given RPM repository and Image Format",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opt.Complete(args); err != nil {
+				log.WithError(err).Error("Cannot complete command")
+				return
+			}
+			if err := opt.Validate(); err != nil {
+				log.WithError(err).Error("Invalid command options")
+				return
+			}
+			if err := opt.Run(); err != nil {
+				log.WithError(err).Error("Runtime error")
+			}
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&opt.Namespace, "namespace", "", "", "namespace to use. Defaults to current namespace.")
+	flags.StringVar(&opt.AccountSecretName, "account-secret", fmt.Sprintf("%s-aws-creds", userName), "secret to use for AWS account")
+	flags.StringVar(&opt.SSHSecretName, "ssh-secret", "libra-ssh", "secret with SSH key")
+	flags.StringVar(&opt.AnsibleImage, "ansible-image", "registry.svc.ci.openshift.org/openshift-cluster-operator/cluster-operator-ansible:latest", "ansible image to use for job")
+	flags.StringVar(&opt.RPMRepositoryURL, "rpm-repository", "", "RPM repository for Origin")
+	flags.StringVar(&opt.RPMRepositorySource, "rpm-repository-source", "https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/releases/openshift/origin/master/origin.repo", "pointer to RPM repository URL")
+	flags.StringVar(&opt.VPCName, "vpc", "", "VPC to use to build the AMI. If left blank, a new VPC will be created")
+	flags.StringVar(&opt.LogLevel, "loglevel", "info", "log level, one of: debug, info, warn, error, fatal, panic")
+	// TODO: It is unclear whether this is needed if we're already providing an RPM repo and ImageFormat
+	flags.StringVar(&opt.Version, "origin-version", "3.11", "Version of origin to provision")
+	flags.StringVar(&opt.ImageFormat, "images", "registry.svc.ci.openshift.org/openshift/origin-v3.11:{component}", "Image format for images to use in the installation")
+	flags.StringVar(&opt.KeyPairName, "keypair", "libra", "Name of AWS keypair to use for created instances")
+	flags.StringVar(&opt.Region, "region", "us-east-1", "AWS region to use to provision cluster")
+	flags.StringVar(&opt.KubeconfigFile, "kubeconfig", "", "Name of a kubeconfig file to create when server is provisioned")
+	flags.StringVar(&opt.ImageID, "ami", "", "[Optional] Image ID to use to provision the cluster. If not specified, a new AMI is created")
+	return cmd
+}
```

Comment:
Yup, we could use this to create a cluster locally... it would require some additional options like not waiting for the cluster to be ready.

Reply:
The high level use case for this is to run this command instead of this call in the e2e test:
https://github.com/openshift/release/blob/master/ci-operator/templates/cluster-launch-e2e.yaml#L191-L193
That call stands up a cluster on GCE (along with a corresponding vm image), ensures that it is running properly and returns success or failure.